### PR TITLE
Vendor to gardener v1.12.3

### PR DESCRIPTION
**Pull request by DockToFuture**
**How to categorize this PR?**

/priority normal
/platform gcp

**What this PR does / why we need it**:
Vendor gardener/gardener@v1.12.3 and update golang verison to `1.15` and alpine image version to `3.12.1`.

**Release note**:

```improvement operator
Golang version is updated to `1.15` and alpine image version is updated to `3.12.1`.
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ############# builder
```
Comment (rfranzke) [inline Dockerfile:1]: Do you also want to update to alpine:3.12.1 now that you are on it?

**Review by rfranzke — COMMENTED** [review]
Thanks!

File: Dockerfile (hunk 1 above)
Comment (DockToFuture) [reply Dockerfile:1]: Sure

**Review by rfranzke — APPROVED** [review]
/lgtm

**Comment by rfranzke** [comment]
Can you update the release note with information about Golang 1.15 and alpine 3.12.1 update?

**Review by prashanth26 — APPROVED** [review]
/lgtm

**Review by dkistner — APPROVED** [review]

**Pull request merged**